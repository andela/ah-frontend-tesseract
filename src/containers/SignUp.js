import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SignUpForm from '../components/auth/SignUpForm';
import { createUser } from '../actions/signupActions';
import { handleSocialResponse } from '../actions';
import SocialButtons from '../components/auth/SocialButtons';
import Popup from '../components/base/Popup';

export class SignUp extends Component {
    createUser = async (data) => {
      await this.props.createUser(data);
    };


    render() {
        return (<div className="container">
            <div className="card">
                <div className="card-panel teal lighten-2">

                    {this.props.user && this.props.user.token ?
                        <span
                            className="white-text">Account created successfully. Check your email for account activation link</span>
                        : (
                            <span className="white-text"> Create Account</span>
                        )
                    }

                </div>

                {!this.props.user.token &&
                    <SignUpForm createUser={this.createUser} user={this.props.user}
                                fetchStatus={this.props.fetchStatus} creation_error={this.props.creation_error}/>
                }

                < DisplaySocialButtons fetchStatus={this.props.fetchStatus} user={this.props.user} handleSocialResponse={this.props.handleSocialResponse} history={this.props.history} />


            </div>
        </div>);
    }
}

const DisplaySocialButtons =(props)=>{

        if (props.fetchStatus)
            return (
            <div className="pop-up">
                <Popup history={props.history} />
            </div>
            );

        else if (!props.user.token)
            return(

                <div className="social">
                    <div id="social-errors"/>
                    <SocialButtons history={props.history}
                                   handleSocialResponse={props.handleSocialResponse}
                    />
                </div>
            );
        else
            return null

};

SignUp.propTypes = {
  createUser: PropTypes.func.isRequired,
  handleSocialResponse: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
};

const mapStateToProps = (state) => {
    return{
        user:state.create_user.user,
        fetchStatus: state.create_user.isFetching,
        creation_error:state.create_user.creation_error
    }
};

export default connect(mapStateToProps, { handleSocialResponse, createUser })(SignUp);
