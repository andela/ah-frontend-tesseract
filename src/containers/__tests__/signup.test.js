/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SignUp from '../SignUp';
import store from '../../store';

Enzyme.configure({ adapter: new Adapter() });

describe('sign-up-form container', () => {
  const createUser = jest.fn();
  const handleSocialResponse = jest.fn();

  it('should render', () => {
    const wrapper = shallow(<SignUp
      createUser={createUser}
      handleSocialResponse={handleSocialResponse}
      store={store}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
