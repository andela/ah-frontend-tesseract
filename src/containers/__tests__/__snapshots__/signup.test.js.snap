// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-up-form container should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SignUp)
    createUser={[MockFunction]}
    handleSocialResponse={[MockFunction]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "createUser": [Function],
      "error": undefined,
      "handleSocialResponse": [Function],
      "loading_status": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
      "user": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "createUser": [Function],
        "error": undefined,
        "handleSocialResponse": [Function],
        "loading_status": false,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "user": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
